### DISCUSSION TITLE
Set as default database automaticaly when i select table in database

### DISCUSSION THREAD

**Discussion by andreysneg** [dbeaver/dbeaver#20569] [Q&A]
Where is option for "Set as default database automaticaly  when i select table in database." ?

This is intuitively correct. And this is how all SQL programs in the world work.

I dont wont 1000 times in day click SET AS DEFAULT  for database.

**Comment by LonwoLonwo** [comment]
Hello @andreysneg 

But what is the reason to set a table database as default if you select a table? Table viewer uses fully qualified names, so you can change data in your table without context changing.

**Comment by andreysneg** [comment]
Reason:

1) In my many years of practice, the SQL code should work with the table that I have selected NOW. i work with mysql/mariadb.
2) Absolutly all DB programs work this way.  

Maybe with other types of bases this is not convenient, so you can do it optionally
You can make options:  SET AS DEFAULT DB OR/AND CONNECTION when table selected.  

Now about >100 times a day I must click CTRL-SHIFT-A .

**Reply by LonwoLonwo** [reply]
Please provide an example of the table viewer using when you need to have only the table database as a default.

**Comment by andreysneg** [comment]
Sql editor .   if i select table in db 'base1' i want use 'base1' for queries. 

This is self-evident.

**Reply by LonwoLonwo** [reply]
But SQL Editor already has a separate connection with a separate database. And if you click on a table in the database navigator tree, a new SQL editor with the correct database context will open.

![2023-07-18 14_30_26-DBeaver Ultimate 23 1 2 - _localhost 5_ Script-55](https://github.com/dbeaver/dbeaver/assets/45152336/8e20f326-c194-4871-87a9-7414869e531a)

**Comment by andreysneg** [comment]
OK. It was giving an error before. Now it's normal.
